langlib/lang.runtime/src/main/ballerina/runtime_errors.bal

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Type for value that can be cloned.
+# This is the same as in lang.value, but is copied here to avoid a dependency.
+type Cloneable readonly|xml|Cloneable[]|map<Cloneable>|table<map<Cloneable>>;
+
+# Error Detail Record
+public type Detail record {|
```

Comment:
Why we need to expose the details record here.  This is available in lang.lib/error

Reply:
@gabilang can we reuse one in `lang.lib/error` or will that create a circular dependency?

Reply:
Since `lang.error` has a dependency on `lang.value`, reusing it will cause cyclic dependency when `lang.runtime` is used by `lang.value`. To avoid cyclic dependency the detail record has been copied at `lang.runtime`.

Reply:
@gabilang what if always load form error module in the `getErrorDetail` method? If that works we don't need to redefine.

Reply:
Yes, error details record in `lang.error` can be used directly by `getErrorDetail` method. Since nothing is used directly by any other modules from the lang.error module it will not give any cyclic dependency. Removed the redefined record and updated the PR.